packages/rich-text/src/component/index.native.js

```diff
@@ -722,51 +722,31 @@ export class RichText extends Component {
 			if ( ! this.isIOS ) {
-				// Aztec performs some html text cleanup while parsing it so, its internal representation gets out-of-sync with the
-				// representation of the format-lib on the RN side. We need to avoid trying to set the caret position because it may
-				// be outside the text bounds and crash Aztec, at least on Android.
-				if ( this.willTrimSpaces( html ) ) {
-					// the html will get trimmed by the cleaning up functions in Aztec and caret position will get out-of-sync.
-					// So, skip forcing it, let Aztec just do its best and just log the fact.
+				// The following regular expression is used in Aztec here:
+				// https://github.com/wordpress-mobile/AztecEditor-Android/blob/b1fad439d56fa6d4aa0b78526fef355c59d00dd3/aztec/src/main/kotlin/org/wordpress/aztec/AztecParser.kt#L656
+				const brBeforeParaMatches = html.match( /(<br>)+<\/p>$/g );
+				if ( brBeforeParaMatches ) {
 					console.warn(
-						'RichText value will be trimmed for spaces! Avoiding setting the caret position manually.'
+						'Oops, BR tag(s) at the end of content. Aztec will remove them, adapting the selection...'
 					);
-					selection = null;
-				} else if (
-					this.props.selectionStart > record.text.length ||
-					this.props.selectionEnd > record.text.length
-				) {
-					console.warn(
-						'Oops, selection will land outside the text, skipping setting it...'
-					);
-					selection = null;
-				} else {
-					// The following regular expression is used in Aztec here:
-					// https://github.com/wordpress-mobile/AztecEditor-Android/blob/b1fad439d56fa6d4aa0b78526fef355c59d00dd3/aztec/src/main/kotlin/org/wordpress/aztec/AztecParser.kt#L656
-					const brBeforeParaMatches = html.match( /(<br>)+<\/p>$/g );
-					if ( brBeforeParaMatches ) {
-						console.warn(
-							'Oops, BR tag(s) at the end of content. Aztec will remove them, adapting the selection...'
-						);
-						const count = (
-							brBeforeParaMatches[ 0 ].match( /br/g ) || []
-						).length;
-						if ( count > 0 ) {
-							let newSelectionStart =
-								this.props.selectionStart - count;
-							if ( newSelectionStart < 0 ) {
-								newSelectionStart = 0;
-							}
-							let newSelectionEnd =
-								this.props.selectionEnd - count;
-							if ( newSelectionEnd < 0 ) {
-								newSelectionEnd = 0;
-							}
-							selection = {
-								start: newSelectionStart,
-								end: newSelectionEnd,
-							};
+					const count = (
+						brBeforeParaMatches[ 0 ].match( /br/g ) || []
+					).length;
+					if ( count > 0 ) {
+						let newSelectionStart =
+							this.props.selectionStart - count;
+						if ( newSelectionStart < 0 ) {
+							newSelectionStart = 0;
+						}
+						let newSelectionEnd =
+							this.props.selectionEnd - count;
+						if ( newSelectionEnd < 0 ) {
+							newSelectionEnd = 0;
 						}
+						selection = {
+							start: newSelectionStart,
+							end: newSelectionEnd,
```

Comment:
Of the three selection checks, this is the only one that triest to do something more intelligent than just setting the selection to `null` if there is a problem, so I decided to leave it in even though I think we could probably remove it as well now.